bdnex/lib/comicrack.py

```diff
@@ -22,20 +24,12 @@ def __init__(self, input_filename=None, comic_info=None):
     def comicInfo_xml_create(self):
         self.logger.info("Create ComicInfo.xml")
 
-        mytree = ET.parse(COMICINFO_TEMPLATE)
-        myroot = mytree.getroot()
-
-        for element in myroot[1][0]:
-            metadata_key = element.attrib['name']
-            if metadata_key in self.comic_info.keys():
-                if element.attrib["type"] == "xs:string":
-                    element.attrib['default'] = str(self.comic_info[metadata_key])
-                elif element.attrib["type"] == "xs:int":
-                    element.attrib['default'] = str(self.comic_info[metadata_key])
-
         tmpdir = tempfile.mkdtemp()
         comic_info_fp = os.path.join(tmpdir, 'ComicInfo.xml')
-        mytree.write(comic_info_fp, encoding='UTF-8', xml_declaration=True)
+
+        data = json.dumps(self.comic_info)
+        tmp_xml = xmlschema.from_json(data, preserve_root=True, schema=schema)
```

Comment:
schema is not defined, I think you meant to have something such as
```python
with open(COMICINFO_TEMPLATE, 'r') as xsd_template:
     schema = xsd_template.read()

     tmp_xml = xmlschema.from_json(data, preserve_root=True, schema=schema)
     ET.ElementTree(tmp_xml).write(comic_info_fp, encoding='UTF-8', xml_declaration=True)
```

However, even with this, this code change is not working unfortunately. Currently, all the values stored in self.comic_info (coming from the json file) are set as strings and converted into str or integer in the deleted code above, note that L34 should have been ```int(self.comic_info[metadata_key])``` instead of ```str(self.comic_info[metadata_key])```. 


So now with this code change, the data coming from the json file is expected to be of the right type but it now breaks against the xsd schema validation, see for example with the number of pages "120": 
```
xmlschema.validators.exceptions.XMLSchemaEncodeError: failed validating '120' with XsdAtomicBuiltin(name='xs:int'):

Reason: '120' is not an instance of <class 'int'>
```

So maybe some refactoring could be done in https://github.com/lbesnard/bdnex/blob/main/bdnex/conf/bdgest_mapping.json to add the type of each key

such as:
```json
{
    "Couleurs": {"value": "Colorist",
                        "type": "str"},
    "Couverture":  {"value": "CoverArtist",
                        "type": "str"},
    "Planches":  {"value": "PageCount",
                         "type": "int"}
}
```

and modify the code accordingly in https://github.com/lbesnard/bdnex/blob/main/bdnex/lib/bdgest.py#L428

thoughts?

Reply:
Just pushed a fix regarding the schema i've forgot... thx for catching it!!
Doesn't the pagecount integer problem haven't been fixed here:
https://github.com/lbesnard/bdnex/pull/5 ?